app/models/patron_request.rb

```diff
@@ -169,11 +169,19 @@ def item_title
   # @return [Array<Folio::Item>] the items in the origin location
   def items_in_location
     @items_in_location ||= bib_data.items.select do |item|
-      if item.effective_location.details['searchworksTreatTemporaryLocationAsPermanentLocation'] == 'true'
-        item.effective_location.code == origin_location_code
-      else
-        item.home_location == origin_location_code
-      end
+      items_in_location_check(item)
+    end
+  end
+
+  # @return [Boolean] if the item is in the location/has a barcode
+  # split out due to robocop
+  def items_in_location_check(item)
+    if barcodes.present? && item.barcode.present?
+      item.barcode.in?(barcodes)
+    elsif item.effective_location.details['searchworksTreatTemporaryLocationAsPermanentLocation'] == 'true'
+      item.effective_location.code == origin_location_code
+    else
+      item.home_location == origin_location_code
```

Comment:
I'm not convinced extracting this method has added clarity or made the method more readable. If anything, it's more confusing now than before.

Maybe extracting `bib_data.items` to a method that applies the barcode filtering would be clearer (and appease rubocop?), or a method that does the filtering on top of `items_in_location` (`selectable_items`, `salient_items`?)

Reply:
I've pushed up https://github.com/sul-dlss/sul-requests/pull/2429 with one of those approaches.. and other than the name, I guess it seems like it'd be ok (and maybe less confusing than having `items_in_location` sometimes further filtered?)

Reply:
Gotcha. Dealer's choice. I fixed it so it should be working with rubocop and tests. I will say that it might be good to have it further filtered by location. There might? be an instance where two different locations have the same barcode?